src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/StarlarkPath.java

```diff
@@ -93,12 +93,12 @@ public StarlarkPath getDirname() {
 
   @StarlarkMethod(
       name = "get_child",
-      doc = "Append the given path to this path and return the resulted path.",
+      doc = "Append the given basename to this path and return the resulting path.",
       parameters = {
-        @Param(name = "child_path", doc = "The path to append to this path."),
+        @Param(name = "child_name", doc = "The basename to append to this path."),
       })
-  public StarlarkPath getChild(String childPath) {
-    return new StarlarkPath(path.getChild(childPath));
+  public StarlarkPath getChild(String childName) {
+    return new StarlarkPath(path.getChild(childName));
```

Comment:
I'm thinking we should maybe just call [`path.getRelative`](https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/vfs/Path.java;l=121;drc=a380a157f658cc392f66f86b6aba340c461dc408) here -- the limitation to only use a basename feels rather arbitrary.

Reply:
I agree that this behavior would be much more useful. Since `get_child` would currently hard fails on such paths, it wouldn't even be a breaking change. Should I submit this change instead?

Reply:
Sounds good to me!

Reply:
Having thought about this some more, what do you think of merging this PR and adding a `get_relative` method that accepts any relative path rather than just a basename? That's because I would usually understand the "children" of a directory to be the entries directly contained in that directory, not a nested subdirectory. We could then mark `get_child` as deprecated if needed.

Reply:
that's also good. I personally don't think it matters much since for anyone expecting "children" to have the narrower meaning, `get_child` with the broader impl would still work. But I won't complain either way :)

Reply:
I found an API design I'm happy for `get_child`. Please take a look.